src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -1232,22 +1235,54 @@ class VariantSampleMatrix[T](val hc: HailContext, val metadata: VariantMetadata,
     filterVariants(p)
   }
 
-  def filterVariantsList(input: String, keep: Boolean): VariantSampleMatrix[T] = {
-    copy(
-      rdd = rdd
-        .orderedLeftJoinDistinct(Variant.variantUnitRdd(sparkContext, input).toOrderedRDD)
-        .mapPartitions({ it =>
-          it.flatMap { case (v, ((va, gs), o)) =>
-            o match {
-              case Some(_) =>
-                if (keep) Some((v, (va, gs))) else None
-              case None =>
-                if (keep) None else Some((v, (va, gs)))
-            }
-          }
-        }, preservesPartitioning = true)
-        .asOrderedRDD
-    )
+  def filterVariantsList(variants: java.util.ArrayList[Variant], keep: Boolean): VariantSampleMatrix[T] =
+    filterVariantsList(variants.asScala.toSet, keep)
+
+  def filterVariantsList(variants: Set[Variant], keep: Boolean): VariantSampleMatrix[T] = {
+    val partitionVariants = variants
+      .groupBy(v => rdd.orderedPartitioner.getPartition(v))
+      .toArray
+      .sortBy(_._1)
+
+    val adjRDD = new AdjustedPartitionsRDD[RowT](rdd,
```

Comment:
I think I can actually do one better: if we sort the par-partition variantSets, then we can do a linear scan on the two iterators.  It didn't seem like we had this out of the box and I got lazy.  I used your implementation for the keep=False case.